.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,72 @@
+name: Benchmarks
+
+on:
+  pull_request:
+    branches: [ '**' ]
+    types: [opened, synchronize, reopened, labeled]
+
+env:
+  PROMSCALE_LOGGING: false
+  SKIP_PROMQL_BENCHMARKS: true
+  golang-version: 1.18.4
+
+jobs:
+  benchmark:
+    if: contains( github.event.pull_request.labels.*.name, 'action:benchmarks')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v3
+
+      - name: Set up Go ${{ env.golang-version }}
+        uses: actions/setup-go@v3.2.1
+        with:
+          go-version: ${{ env.golang-version }}
+        id: go
+
+      - name: Benchmark current PR
+        run: go test -bench=. -run=^$ -benchmem -cpu=1 -count=10 -timeout=6h ./... | tee current.txt
+
+      - name: Upload results
+        uses: actions/upload-artifact@v3
+        with:
+          name: current
+          path: current.txt
+
+      - name: Checkout master branch
```

Comment:
I'm not sure how easy it is, but can we run PR & master branch benchmarking in parallel?

Reply:
If you define the benchmarks in separate jobs, then they run on different github runners (in parallel). But if they run on different runners they have different performance characteristics, so we can't compare them. If they run on parallel on the same runner, they will contend for resources, so we won't be able to compare the results.

Reply:
> If you define the benchmarks in separate jobs, then they run on different github runners (in parallel). But if they run on different runners they have different performance characteristics, so we can't compare them
I get your point. Isn't a noisy neighbour can still affect sequential steps? IIUC, in github's terms runner refers to a VM not a physical machine.

> If they run on parallel on the same runner, they will contend for resources, so we won't be able to compare the results.
I'm not aware that we can run parallel jobs in the same runner.

Reply:
That was the first approach. The problem was that, parallel workflows lead to being scheduled on different machines, which introduced throttling results.
To see, please refer to this comment https://github.com/timescale/promscale/pull/1505#issuecomment-1190040727 and see the comment history towards the start, when benchmarks were running in parallel.

Hence, now the jobs run serially, but on same machine, which gives some reduction of noise in the results.

Reply:
> Isn't a noisy neighbour can still affect sequential steps

Yes, definitely. I think that we're going to have to see how reliable these results are in practice. Based on [this](https://github.com/timescale/promscale/pull/1511#issuecomment-1191349375), there is still some noise... If the results aren't consistent enough then we might need to use a dedicated machine for this.

Reply:
I think the easy way to make a decision on parallel jobs vs serial jobs, is to see the throttling of results. I saw that throttling was reduced when the jobs ran on serially (due to same runner), than when in parallel